cmd/clusterctl/client/get_kubeconfig.go

```diff
@@ -56,5 +61,15 @@ func (c *clusterctlClient) GetKubeconfig(options GetKubeconfigOptions) (string,
 		options.Namespace = currentNamespace
 	}
 
-	return clusterClient.WorkloadCluster().GetKubeconfig(options.WorkloadClusterName, options.Namespace)
+	// system kubeconfig is explicitly asked for
+	if !options.UserKubeconfig {
+		return clusterClient.WorkloadCluster().GetKubeconfig(options.WorkloadClusterName, options.Namespace)
+	}
+	kubeconfig, err := cluster.GetUserKubeconfig(clusterClient.Proxy(), options.WorkloadClusterName, options.Namespace)
+	if err != nil {
+		// fallback on the workload cluster kubeconfig
+		klog.Warningf("failed to fetch the user kubeconfig; fetching the system kubeconfig instead")
```

Comment:
Can you please switch to using the logger from `"sigs.k8s.io/cluster-api/cmd/clusterctl/log"`?

Ref for using it: https://github.com/ykakarap/cluster-api/blob/1da5d29c9ec54b9dcd6bb261b4cae8864a9d2734/cmd/clusterctl/client/init.go#L97